### PR TITLE
Fix Mbed TLS build for Android

### PR DESCRIPTION
**Pull request by ahmedyarub**
**Describe the pull request**
Allows using VCPKG toolchain + Android toolchain. Otherwise, we would not be able to use Android toolchain together with VCPKG building Mbed TLS. After this, SDK building should work on VCPKG. It also fixes the check for the environment (Windows) to make it compatible with cross-compilation.
This change can be easily extended to enable building with VCPKG + iOS/Wasm or any other cross-platform build.

*Check all that applies:*
- [x] Did a review by yourself.
- [ ] Added proper tests to cover this PR. (If tests are not applicable, explain.)
- [x] Checked if this PR is a breaking (APIs have been changed) change.
- [x] Checked if this PR will _not_ introduce cross-platform inconsistent behavior.
- [x] Checked if this PR would require a ReadMe/Wiki update.

Check which platforms you have built SDK on to verify the correctness of this PR.
- [x] Linux
- [x] Windows
- [x] Android
- [x] MacOS
- [ ] IOS
- [ ] Other Platforms

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by JackBoosY** [comment]
Please run command `vcpkg x-add-version --all` then commit the changes.

Thanks.

**Comment by ahmedalnuaimi** [comment]
> Please run command `vcpkg x-add-version --all` then commit the changes.
> 
> Thanks.

Done

**Comment by ahmedyarub** [comment]
> 
> 
> Please run command `vcpkg x-add-version --all` then commit the changes.
> 
> Thanks.

Done

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
@ahmedyarub I found that some of the items in your comment have not yet been implemented. Is this PR ready?

**Comment by ahmedyarub** [comment]
> 
> 
> @ahmedyarub I found that some of the items in your comment have not yet been implemented. Is this PR ready?

I`m sorry which comments exactly? If you are talking about Wasm and iOS support then that would be done in a separate PR during the next week.

**Comment by JackBoosY** [comment]
Test all features successfully on `x86-windows` and `x64-windows-static`.

**Comment by ahmedyarub** [comment]
Fixed according to https://vcpkg.readthedocs.io/en/latest/users/android/

**Comment by JackBoosY** [comment]
@ahmedyarub Ready for review now?

**Comment by ahmedyarub** [comment]
> 
> 
> @ahmedyarub Ready for review now?

Yes it is

File: ports/mbedtls/portfile.cmake [hunk 1]
```diff
@@ -3,28 +3,33 @@ vcpkg_fail_port_install(ON_TARGET "uwp")
...
-    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
-    FEATURES
-    pthreads ENABLE_PTHREAD
+        OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+        FEATURES
+        pthreads ENABLE_PTHREAD
 )
 
+if (VCPKG_TARGET_IS_ANDROID)
+    set(EXTRA_ARGS "-DMBEDTLS_FATAL_WARNINGS=FALSE")
```
Comment (ras0219) [inline ports/mbedtls/portfile.cmake:22]: Can this be applied always? Our policy is to not enable warnings as errors in package builds.

File: ports/mbedtls/portfile.cmake [hunk 2]
```diff
@@ -3,28 +3,33 @@ vcpkg_fail_port_install(ON_TARGET "uwp")
 set(VCPKG_LIBRARY_LINKAGE static)
 
 vcpkg_from_github(
-    OUT_SOURCE_PATH SOURCE_PATH
-    REPO ARMmbed/mbedtls
-    REF 523f0554b6cdc7ace5d360885c3f5bbcc73ec0e8 # mbedtls-2.24.0
-    SHA512 1ce78f34e8d87c2ce0454e0a08f4c6e5b3129d4b24cfa44162af21c2e8b5dc7feabf849e4fa547ce3781b5ce11aaf675cfed47412bae40091fbdd87bbcdbee07
-    HEAD_REF master
-    PATCHES
+        OUT_SOURCE_PATH SOURCE_PATH
```
Comment (ras0219) [inline ports/mbedtls/portfile.cmake:6]: Could you please restore the formatting of this file to the original 4 spaces?

**Review by ras0219 — CHANGES_REQUESTED** [review]
LGTM once my two comments are addressed!

File: ports/mbedtls/portfile.cmake (hunk 2 above)
Comment (ahmedyarub) [reply ports/mbedtls/portfile.cmake]: Done

**Review by JackBoosY — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Already merged from master.

**Comment by ahmedyarub** [comment]
This should be ready for last review and merge.

**Pull request merged**